WordPress/gutenberg
# SiteTitle: Add bold and italic toolbar buttons for convenience

- Closes: #45854

## What?

Add Font face control to the SiteTitle toolbar where users can use `Bold` and `Italic` text styles:



<img width="449" alt="Screenshot 2023-02-24 at 14 26 45" src="https://user-images.githubusercontent.com/779993/221202947-adbfaaaa-e1e3-4b61-9fa6-bfef091a6f98.png">


## Why?

The `SiteTitle` style options are difficult to access. I propose adding the bold and italic buttons to the toolbar allowing users to edit the text style easily.
In addition, we improve the accessibility of the block by allowing the shortcuts `Ctrl + B` to make the text bold and `Ctrl + I` to make it italic.

## How?

I've added two new Toolbar buttons to the SiteTitle block. I'm using the `attribtues.style` as the source of truth of the bold and style state. For this reason, the new buttons and the Appearance Selector are in sync. Check the screencast to see it in action.

## Testing Instructions




1. Go to Appearance > Editor.
2. Click on `Edit`.
3. Select your `SiteTitle` block.
4. Observe that there are a new Typography button on the Toolbar, click it and observe the dropdown with the Bold and Italic options.
5. Click on the Bold button.
6. Observe the text transforms to bold.
7. Click on the Italic button.
8. Observe the text transforms to italic.
9. Observe that the Appearance select in the inner settings has changed too. The toolbar and the select are in sync.
10. Save the template.
11. Observe the Frontend of the site shows the correct styles.


### Testing Instructions for Keyboard

1. Select the `SiteTitle` block
8. Press `Ctrl + B` or `⌘ + B`
12. Observe the text gets bold.
13. Press `Ctrl + I` or `⌘ + I`
14. Observe the text gets italic.

## Screenshots or screencast 

| **Before** | **After** |
|---|---|
| <img width="1178" alt="site-title-before" src="https://user-images.githubusercontent.com/779993/207203618-c669f32a-7566-4b17-8190-e9c221e327e9.png"> | <img width="1262" alt="Screenshot 2023-02-24 at 14 38 04" src="https://user-images.githubusercontent.com/779993/221205920-70a22e40-a829-484c-8024-c6f95c6cedcc.png"> |







https://user-images.githubusercontent.com/779993/220179387-4db637e5-9a68-44e6-b6fb-901e0b50c9a0.mp4
> The Typography icon was updated here https://github.com/WordPress/gutenberg/pull/46489#issuecomment-1443630583
